src/dune_rules/foreign.ml

```diff
@@ -159,9 +180,20 @@ module Stubs = struct
                (foreign_library ...) stanza."
           ]
     in
-    { loc; language; names; flags; include_dirs; extra_deps }
+    let () =
+      match mode with
+      | Some _ when for_library ->
+        User_error.raise ~loc:loc_mode
+          [ Pp.textf "The field \"mode\" is not available for foreign_libraries"
```

Comment:
"foreign_libraries" isn't a stanza. Should be singular or just "foreign libraries"

Reply:
802a73fda1937ccfb6b453e6ae364aef99593821